src/components/InjectedComponents/DecryptedPost/DecryptedPost.tsx

```diff
@@ -16,11 +16,41 @@ import { DecryptPostFailedProps, DecryptPostFailed } from './DecryptPostFailed'
 import { DecryptedPostDebug } from './DecryptedPostDebug'
 import { usePostInfoDetails } from '../../DataSource/usePostInfo'
 import { asyncIteratorWithResult } from '../../../utils/type-transform/asyncIteratorHelpers'
+import { getActivatedUI } from '../../../social-network/ui'
+
+function progressReducer(
+    state: { key: string; progress: SuccessDecryption | FailureDecryption | DecryptionProgress }[],
+    {
+        key,
+        progress,
+    }: {
```

Comment:
I have a coding style suggestion but it will not block the PR to be merged.

When deconstructing parameters, I suggest to do:

```ts
function f(opts: { key: string, progress }) {
    const { key, progress } = opts
}
```

The reason is that, in your case, these statements use 8 lines, and in the style I mentioned, it will only cause about 2~4 lines and make the code easier to read.